examples/basic_example/graph_client_basic_example.go

```diff
@@ -31,9 +34,24 @@ func main() {
 	hostList := []nebula.HostAddress{hostAddress}
 	// Create configs for connection pool using default values
 	testPoolConfig := nebula.GetDefaultConf()
+	testPoolConfig.UseHTTP2 = useHTTP2
+
+	var sslConfig *tls.Config
+	if useSSL {
+		var err error
+		sslConfig, err = nebula.GetDefaultSSLConfig(
+			"./nebula-docker-compose/secrets/test.ca.pem",
+			"./nebula-docker-compose/secrets/test.client.crt",
+			"./nebula-docker-compose/secrets/test.client.key",
+		)
+		if err != nil {
+			log.Fatal("Fail to create ssl config")
+		}
+		sslConfig.InsecureSkipVerify = true
+	}
 
 	// Initialize connection pool
-	pool, err := nebula.NewConnectionPool(hostList, testPoolConfig, log)
+	pool, err := nebula.NewSslConnectionPool(hostList, testPoolConfig, sslConfig, log)
```

Comment:
it is more friendly for users to retain the basic connection pool example, and `add` an example for ssl pool instead of replacing the basic one?

Reply:
If `sslConfig` is nil, no ssl is used.

```go
func NewConnectionPool(addresses []HostAddress, conf PoolConfig, log Logger) (*ConnectionPool, error) {
	return NewSslConnectionPool(addresses, conf, nil, log)
}
```

Reply:
the usage is no problem, I mean there's no example for no ssl ConnectionPool.
So please remain the old NewConnectionPool example, and `add` your NewSslConnectionPool rather than `replace` it.

Reply:
I am wondering whether we need to keep both `NewConnectionPool` and `NewSSLConnectionPool`

Reply:
> I am wondering whether we need to keep both `NewConnectionPool` and `NewSSLConnectionPool`

good question, there's no need to keep two functions just for one ssl parameter.  I'll refactor it Just like https://github.com/vesoft-inc/nebula-java/blob/c39481ba2c264d183a5db89195b60bbf59fe0677/client/src/main/java/com/vesoft/nebula/client/graph/NebulaPoolConfig.java#LL46C3-L60C6

Reply:
> the usage is no problem, I mean there's no example for no ssl ConnectionPool. So please remain the old NewConnectionPool example, and `add` your NewSslConnectionPool example rather than `replace` it.

This example also supports no ssl. What do you want to achieve?

Reply:
> > I am wondering whether we need to keep both `NewConnectionPool` and `NewSSLConnectionPool`
> 
> good question, there's no need to keep two functions just for one ssl parameter. I'll refactor it Just like https://github.com/vesoft-inc/nebula-java/blob/c39481ba2c264d183a5db89195b60bbf59fe0677/client/src/main/java/com/vesoft/nebula/client/graph/NebulaPoolConfig.java#LL46C3-L60C6

I don't think this is what this PR needs to do. And the change will also lead to incompatibility, I personally do not recommend such a change.